### PR TITLE
Feature/implement header section

### PR DESCRIPTION
**Pull request by HoangLiarPhuongLe**
- Create Navbar Component
- Create Button, ButtonIcon Components and Storybook
- Create Header Component
image

### CODE REVIEW

File: practice-one/src/components/button-icon/index.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import type { Meta, StoryObj } from '@storybook/react'
```
Comment (huyenpham-agilityio) [inline practice-one/src/components/button-icon/index.stories.tsx]: please check folder naming

File: practice-one/src/components/button-icon/index.tsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
+const ButtonIcon = (props: ButtonIconProps) => {
```
Comment (huyenpham-agilityio) [inline practice-one/src/components/button-icon/index.tsx]: destructure props here instead

ex: const ButtonIcon = ({source, alt, primary}: ButtonIconProps) => {

File: practice-one/src/components/button-icon/index.tsx [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    <button
+      className={`cursor-pointer rounded-full p-2 hover:shadow-sm ${primary ? 'bg-red-500' : ''}`}
+    >
+      <img src={source} alt={alt} />
+    </button>
+  )
+}
+
+type ButtonIconProps = {
+  primary?: boolean
```
Comment (huyenpham-agilityio) [inline practice-one/src/components/button-icon/index.tsx]: why use `primary`, how about if you have many types for one component?

File: practice-one/src/components/button/index.tsx [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import { ReactNode } from 'react'
+
+const Button = (props: ButtonProps) => {
+  const { children, primary = false, size = 'medium' } = props
```
Comment (huyenpham-agilityio) [inline practice-one/src/components/button/index.tsx]: destructure at props

File: practice-one/src/components/button-icon/index.tsx [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  return (
+    <button
+      className={`cursor-pointer rounded-full p-2 hover:shadow-sm ${primary ? 'bg-red-500' : ''}`}
+    >
+      <img src={source} alt={alt} />
+    </button>
+  )
+}
+
+type ButtonIconProps = {
```
Comment (huyenpham-agilityio) [inline practice-one/src/components/button-icon/index.tsx]: props type should be defined before component

File: practice-one/src/components/navbar/index.tsx [hunk 6]
```diff
@@ -0,0 +1,20 @@
+const Navbar = (props: { list: { to: string; content: string }[] }) => {
+  const navList = props.list.map((item) => (
+    <li
+      key={item.content}
+      className="cursor-pointer capitalize hover:font-bold hover:text-blue-400"
+    >
+      <a href={item.to}>{item.content}</a>
+    </li>
```
Comment (huyenpham-agilityio) [inline practice-one/src/components/navbar/index.tsx]: split to NavbarItem component

File: practice-one/src/components/navbar/index.tsx [hunk 7]
```diff
@@ -0,0 +1,20 @@
+const Navbar = (props: { list: { to: string; content: string }[] }) => {
```
Comment (huyenpham-agilityio) [inline practice-one/src/components/navbar/index.tsx]: create type for props instead

File: practice-one/src/components/button-icon/index.tsx (hunk 3 above)
Comment (HoangLiarPhuongLe) [reply practice-one/src/components/button-icon/index.tsx]: Because there are just two types for this component

Comment (huyenpham-agilityio) [reply practice-one/src/components/button-icon/index.tsx]: should use `variant` then pass type primary or secondary...

Comment (HoangLiarPhuongLe) [reply practice-one/src/components/button-icon/index.tsx]: I fixed @huyenpham-agilityio . Please help me check. Thanks!

**Review by huyenpham-agilityio — APPROVED** [review]

**Pull request merged**